core/src/main/scala-2/sttp/tapir/internal/SchemaEnumerationMacro.scala

```diff
@@ -12,4 +14,25 @@ object SchemaEnumerationMacro {
       new sttp.tapir.macros.CreateDerivedEnumerationSchema(Validator.derivedEnumeration)
     """)
   }
+
+  def derivedEnumerationValue[T: c.WeakTypeTag](c: blackbox.Context): c.Expr[Schema[T]] = {
+    import c.universe._
+
+    val Enumeration = typeOf[scala.Enumeration]
+    val SchemaAnnotations = typeOf[sttp.tapir.internal.SchemaAnnotations[_]]
+
+    val weakTypeT = weakTypeOf[T]
+    val owner = weakTypeT.typeSymbol.owner
+
+    if (!(owner.asClass.toType <:< Enumeration)) {
+      c.abort(c.enclosingPosition, "Can only derive Schema for values owned by scala.Enumeration")
+    } else {
+
+      val enumeration = TermName(weakTypeT.toString.split("\\.").dropRight(1).last)
```

Comment:
Yep, I tried that but it results in compilation error - `not found: value sttp.tapir.generic.Countries` which is kind of surprising, since that class is available on the macro call site (and it works in scala3).

Reply:
hm maybe we would have to generate another AST - sth like a `Select(prefix, TermName(className))`?

Reply:
what should be the `prefix` here?

Reply:
`weakTypeT.toString.split("\\.").dropRight(1).init`

Reply:
but `Select` first argument is a `Tree`

Reply:
well you'd have to print an AST of a fully qualified class reference and see how it looks :)